src/collapse/mod.rs

```diff
@@ -1,63 +1,39 @@
-pub mod bpftrace;
-pub mod perf;
+mod bpftrace;
+mod perf;
 
+pub use bpftrace::Bpftrace;
+pub use perf::{Perf, PerfOptions};
+
+use std::fs::File;
 use std::io;
 use std::path::Path;
-use std::fs::File;
 
 const READER_CAPACITY: usize = 128 * 1024;
 
-/// Type that knows how to collapse formats from various frontends (e.g. bpftrace and perf)
-/// into output that can be subsequently ingested by `inferno-flamegraph`.
-pub struct Collapser {
-    frontend: Frontend,
-}
-
-impl Collapser {
-    /// Constructs a new `Collapser`.
-    pub fn new(frontend: Frontend) -> Self {
-        Collapser { frontend }
-    }
+pub trait Frontend {
```

Comment:
I'd like to see documentation on this trait and its methods, since authors of new collapsers will have to implement it :)

Reply:
👍